urllib3/util/ssl_.py

```diff
@@ -17,6 +18,23 @@
 
 import errno
 import warnings
+import hmac
+
+if hasattr(hmac, 'compare_digest'):
+    _const_compare_digest = hmac.compare_digest
+else:
+    def _const_compare_digest(a, b):
+        """
+        Compare two digests of equal length in constant time.
+
+        Returns True if the digests match, and False otherwise.
+        """
+        result = abs(len(a) - len(b))
+        for l, r in six.moves.zip(bytearray(a), bytearray(b)):
```

Comment:
Python 2's zip will eagerly create a list, which is unnecessary here. I don't think six.moves.zip should be a problem, and it definitely seems to exist in the packaged version of six.py.